common/src/main/java/com/microsoft/identity/common/internal/broker/InstallCertActivityLauncher.java

```diff
@@ -0,0 +1,167 @@
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+package com.microsoft.identity.common.internal.broker;
+
+import android.app.Activity;
+import android.content.Intent;
+import android.os.Bundle;
+
+import androidx.activity.result.ActivityResultCallback;
+import androidx.activity.result.ActivityResultLauncher;
+import androidx.activity.result.contract.ActivityResultContract;
+import androidx.activity.result.contract.ActivityResultContracts;
+import androidx.annotation.NonNull;
+import androidx.appcompat.app.AppCompatActivity;
+
+import com.microsoft.identity.common.PropertyBagUtil;
+import com.microsoft.identity.common.internal.logging.Logger;
+import com.microsoft.identity.common.java.exception.ClientException;
+import com.microsoft.identity.common.java.util.StringUtil;
+import com.microsoft.identity.common.java.util.ported.LocalBroadcaster;
+import com.microsoft.identity.common.java.util.ported.PropertyBag;
+
+
+import lombok.experimental.Accessors;
+
+/**
+ * Base activity used to launch a Install WPJ Certificate request,
+ * and get the result from the Activity using the API {@link AppCompatActivity#registerForActivityResult(ActivityResultContract, ActivityResultCallback)}
+ */
+@Accessors(prefix = "m")
+public final class InstallCertActivityLauncher extends AppCompatActivity {
+    private static final String TAG = InstallCertActivityLauncher.class.getSimpleName();
+    private static final String INSTALL_CERT_INTENT_STARTED = "broker_intent_started";
+    private static final String INSTALL_CERT_INTENT = "install_cert_intent";
+    private static final String INSTALL_CERT_BROADCAST_ALIAS = "install_cert_broadcast_alias";
+    private Intent mInstallCertificateIntent;
+    private Boolean mInstallCertificateIntentStarted = false;
+    private Boolean mInstallCertificateResultReceived = false;
+
+    // installCertActivityResultLauncher creates an ActivityResultLauncher<Intent>
+    // to process the result form the install cert activity
+    final ActivityResultLauncher<Intent> installCertActivityResultLauncher = registerForActivityResult(
+            new ActivityResultContracts.StartActivityForResult(),
+            activityResult -> {
+                final String methodTag = TAG + "#installCertActivityResultLauncher";
+                Logger.info(methodTag, "Result received from Broker, Result code: " + activityResult.getResultCode());
+                final Intent intentWithData = activityResult.getData();
+                final PropertyBag resultBag;
+                if (intentWithData == null || intentWithData.getExtras() == null) {
+                    Logger.error(methodTag, "Certificate installation failed with an empty response", null);
+                    resultBag = new PropertyBag();
```

Comment:
Do we want to put some error in the property bag here?

Reply:
The error should be returned with the resultBag.